### PR TITLE
modularize tests

### PR DESCRIPTION
**Pull request by tejasmanohar**
test/application.js was a monster

### CODE REVIEW

**Comment by tejasmanohar** [comment]
hmm. any ideas on why tests and coverage checks are passing? all tests pass locally w/ `npm test` but not w/ `make test-travis`, and they were just re-arranged so not quite sure. i think it's something to do w/ the request testing and how i removed some of the require's when decoupling the "application" tests but not really sure.

**Comment by jonathanong** [comment]
yeah i've seen that before locally. not sure.

**Comment by jonathanong** [comment]
screen shot 2015-10-11 at 9 24 41 pm

this failed for me though

File: test/context/toJSON.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 
 'use strict';
 
-const context = require('../context');
+const context = require('.');
```
Comment (jonathanong) [inline test/context/toJSON.js]: `require('.')` might be a special cast for `process.cwd()`... not sure if that's still the case

Comment (tejasmanohar) [inline test/context/toJSON.js]: hmm, as far as i can tell `'.'` is just shorthand for `'./'` in node's directory system... but may be wrong.

**Comment by tejasmanohar** [comment]
@jonathanong hmm, tests pass locally on my end w/ `npm test` - is that what you're running them with, too? also, verify `git rev-parse HEAD` to see we're both on the same commit 4d2f88cbe263413c89a0c3097465cf4a141598d4?

on your `process.cwd()` note... does replacing `require('.')` w/ `require('./')` or `require('./index.js')` help with your tests? i would imagine if `'.'` was being treated as `process.cwd()` then, this would fail saying it can't find a module `index.js` in the root dir or resort to `lib/application.js` since that's the `main` in `package.json` which would make all the context tests fail though...

File: test/context/toJSON.js (hunk 1 above)
Comment (targos) [inline test/context/toJSON.js]: > hmm, as far as i can tell '.' is just shorthand for './' in node's directory system... 

That is correct.

**Comment by tejasmanohar** [comment]
will retry and commit more frequently to see where/when travis fials

**Pull request closed without merging**